### ISSUE TITLE
Checkbox state announced before the label.

### ISSUE THREAD

**Issue by audiogamer22** [nvaccess/nvda#17242]
### Is your feature request related to a problem? Please describe.
No.
### Describe the solution you'd like
I'd like to either have a combo box or checkbox in the NVDA settings that allows the checkbox state to be announced before the label is read. This is great if you already have an idea of the programs layout but can't remember how many tabs it takes to get to a place.

I've found an addon that does this and find it works great for what I need but i think adding it as a feature in NVDA will help those that need/like it.
### Describe alternatives you've considered
a little ticking/unticking sound before the lable might work just as well.
### Additional context
I understand why screen readers read things this way, but  I think getting the state of the checkboxes and perhaps even radio buttons before hearing the label is quicker, or it seams that way to me at least.

I'd like to hear what others think as I think this will be an  interesting discussion on the behaviour of screen readers and what we personally prefer.

**Comment by XLTechie** [comment]
I for one, certainly agree with this feature. Especially with badly designed web pages out there, that put the whole terms of service or similar, in the label of the checkbox.

**Comment by audiogamer22** [comment]
Exactly. If the users don't like it,, they can just have it how NVDA usually is.

**Comment by Adriani90** [comment]
Just to mention, this depends on how you navigate. If you use arrow keys in browse mode, the state is announced before the label.
If you use tab or shift+tab or the quick navigation key, the label is announced first. So to a certain extent users can influence this by the navigation patern.

**Comment by SaschaCowley** [comment]
Special case of #7232

**Comment by XLTechie** [comment]
Ah yes, quite. I knew it was a duplicate, but haven't had time to search for the base issue.

**Comment by TechHorseG** [comment]
I hope that I am not posting this in the wrong thread. Please let me know if there is a more suitable place for these comments.

I don't have a single one-size-fits-all preference for whether the form control state should be read before or after its label.

If I am trying to locate a particular setting then I want to hear the label first, and not be slowed down by having to wait for 'checkbox not checked' / 'toggle button pressed' etc. before each item. That would irritate me, especially if there are a lot of settings to navigate through to find the one that I want.

At other times I just want to check what the setting's state is and do not want to have to wait for a long label or description. An example  of a bad offender here is when checking whether an application has been set to run as an administrator. Before hearing the checkbox state I have to wait for 'Run as administrator. This option allows you to run this shortcut as an administrator, while protecting your computer from unauthorised activity.'

This is where in browse mode at least, I appreciate the fact that I can change the information order on the fly just by changing the navigation method. Pressing tab or F sees the label being read first, but I can quickly check form control states just by arrowing up and down instead, as this of course sees the state read first. I frequently make use of the different browse mode navigation styles to quickly get the information that I want at that moment.

So on that note, if a form control reading order setting is to be offered, then at least also allow the current status quo of different orders for different navigation styles to be retained. Otherwise we'd have to keep opening and changing settings to have label read first, or state read first, depending on what is most useful at the time.

Either that, or please offer one gesture to read only the form state, another gesture to read only the label, and a third to allow for the reading order to be toggled on the fly.

That way I could have my generally preferred label-first reading order, but also be able to quickly check a single setting's state by invoking the 'read state only' gesture. Or if I am already familiar with the labels and just want to quickly check that a group of settings are enabled for example, then I could invoke the 'flip reading order' gesture to hear either 'checkbox checked' or 'checkbox not checked' read first for this session, then flip back afterwards.
